### DISCUSSION TITLE
French words and check() function

### DISCUSSION THREAD

**Discussion by mathieu-desgagnes** [ateucher/pkg-dev-dfo-2024-11#31] [General]
Hey!
I'm excited to try my new knowledge on older code! But I got that warning while runing check()...

❯ checking code files for non-ASCII characters ... WARNING
  Found the following file with non-ASCII characters:
    R/lib-summary.R
  Portable packages must use only ASCII characters in their R code and
  NAMESPACE directives, except perhaps in comments.
  Use \uxxxx escapes for other characters.
  Function 'tools::showNonASCIIfile' can help in finding non-ASCII
  characters in files.

0 errors ✔ | 1 warning ✖ | 0 notes ✔

To reproduce this warning use this small modification to our function:
```r
lib_summary <- function() {
  pkgs <- utils::installed.packages()
  pkg_tbl <- table(pkgs[,'LibPath'])
  pkg_df <- as.data.frame(pkg_tbl, stringsAsFactors = FALSE)
  names(pkg_df) <- c('Livrary', 'n_packages')
  print("C'est réussi")
  pkg_df
}
```

Lots of French words are needed to produce plots for publication in both languages, so I'm really interested to know the best way to avoid this warning.

**Comment by ateucher** [comment]
Hi @mathieu-desgagnes - great question! Accented characters can be included in R code by encoding them with their unicode strings - starting with `\U`. You can look up unicode equivalents on many websites, or you can use the (slightly cumbersome) `iconv()` function in R:

``` r
iconv("é", to = "ASCII", sub = "Unicode")
#> [1] "<U+00E9>"
```

You have to change the format from `<U+xxxx>` to `\U{xxxx}>`, so it would look like this:

``` r
lib_summary <- function() {
  pkgs <- utils::installed.packages()
  pkg_tbl <- table(pkgs[, "LibPath"])
  pkg_df <- as.data.frame(pkg_tbl, stringsAsFactors = FALSE)
  names(pkg_df) <- c("Livrary", "n_packages")
  print("C'est r\U{00E9}ussi")
  pkg_df
}

lib_summary()
#> [1] "C'est réussi"
#>                                                                Livrary
#> 1 /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/library
#> 2                              /Users/andy/Library/R/arm64/4.4/library
#>   n_packages
#> 1         29
#> 2        376
```

This does pass check because all of the characters are ascii, but when actually printed are displayed in the unicode form
